File: Algorithms/7. Reverse Integer/kotlin/solution.kt
class Solution {
    fun reverse(x: Int): Int {
        var result = 0L
        var tmp = x
        while (tmp != 0) {
            result = result * 10 + tmp % 10
            tmp /= 10
        }
        return if (result > Integer.MAX_VALUE || result < Integer.MIN_VALUE) 0 else result.toInt()
    }
}

File: Algorithms/7. Reverse Integer/python/solution.py
class Solution:
    def reverse(self, x: int) -> int:
        """
        :type x: int
        :rtype: int
        """
        if x > 0:
            result = int(str(x)[::-1])
        else:
            result = -1 * int(str(-x)[::-1])

        if result > (2**31 - 1) or result < -(2**31):
            result = 0
        return result
